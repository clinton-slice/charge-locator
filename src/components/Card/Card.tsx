import { StyleSheet, Text, View, TouchableOpacity } from "react-native";
import Button from "../Button";

interface Props {
  title: string;
  address: string;
  town: string;
  stateOrProvince: string;
  ocmNumber: string;
  postcode: string;
  onNavigate: () => void;
  onStartCharge: () => Promise<void>;
}

export const Card = ({
  ocmNumber,
  title,
  address,
  stateOrProvince,
  town,
  postcode,
  onNavigate,
  onStartCharge,
}: Props) => {
  return (
    <View style={[styles.container, styles.shadowProp]}>
      <View style={styles.contentContainer}>
        <Text style={styles.header}>{title}</Text>
        <Text>OCM-{ocmNumber}</Text>
        <Text>{address}</Text>
        <Text>
          {town}, {stateOrProvince} {postcode}
        </Text>
      </View>
      <View style={styles.buttonContainer}>
        <Button
          onPress={onNavigate}
          label="Navigate"
          style={styles.navButton}
        />
        <Button onPress={onStartCharge} label="Start Charging" />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    borderRadius: 8,
    borderWidth: 1,
    borderColor: "#bfbfbf",
    display: "flex",
    justifyContent: "space-between",
    backgroundColor: "white",
    width: "100%",
    marginVertical: 10,
  },
  contentContainer: {
    padding: 16,
    borderBottomColor: "#bfbfbf",
    borderBottomWidth: 1,
  },
  buttonContainer: {
    display: "flex",
    flexDirection: "row",
  },
  navButton: {
    borderRightWidth: 1,
    borderRightColor: "#bfbfbf",
  },
  shadowProp: {
    shadowColor: "#171717",
    shadowOffset: { width: -2, height: 4 },
    shadowOpacity: 0.2,
    shadowRadius: 3,
  },
  header: {
    fontSize: 14,
    fontWeight: "bold",
  },
});
