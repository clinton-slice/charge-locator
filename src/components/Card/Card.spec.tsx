import React from "react";
import { render, screen, fireEvent } from "@testing-library/react-native";
import Card from ".";

describe("Card", () => {
  const mockOnCharge = jest.fn();
  const mockOnNavigate = jest.fn();
  const component = (
    <Card
      title="Title"
      address="123 Main street"
      town="Ottawa"
      stateOrProvince="Ontario"
      postcode="A1A Y67"
      ocmNumber="9000"
      onStartCharge={mockOnCharge}
      onNavigate={mockOnNavigate}
    />
  );
  it("renders the correct message", () => {
    const { container } = render(component);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("calls `onNavigate` callback when  clicked", () => {
    render(component);
    fireEvent.press(screen.getByText("Navigate"));
    expect(mockOnNavigate).toHaveBeenCalled();
  });
  it("calls `onStartCharge` callback when button is clicked", () => {
    render(component);
    fireEvent.press(screen.getByText("Start Charging"));
    expect(mockOnCharge).toHaveBeenCalled();
  });
});
