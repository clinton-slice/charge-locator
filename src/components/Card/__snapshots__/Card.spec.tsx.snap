// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders the correct message 1`] = `undefined`;
